### PR TITLE
Return assembler object in AssemblyStack

### PR DESCRIPTION
**Pull request by axic**
Solves the problem mentioned in https://github.com/ethereum/solidity/pull/2233#discussion_r118927275

### CODE REVIEW

**Comment by chriseth** [comment]
I'm not really sure which problem this PR solves. I think we should find time to talk tomorrow :-)

**Comment by axic** [comment]
@chriseth the problem is in the description :)

**Review by chriseth — COMMENTED** [review]
Please rebase.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
I think we agreed to: (comments in #2233)
- keep `assemble()`, but return an empty object if it is not supported by the backend
- introduce `assembly()` to return a text assembly or empty string if not supported by the backend
- rename `AssemblyStack` to `BackendStack`

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -1096,7 +1101,18 @@ bool CommandLineInterface::assemble(
 			cerr << "Unknown exception while assembling." << endl;
 			return false;
 		}
-		cout << stack.print() << endl;
+
+		cout << endl << "Binary representation:" << endl;
+		if (object.bytecode)
+			cout << object.bytecode.get()->toHex() << endl;
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: You do not need to call `get()` if you use `->`.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
@chriseth so this means we are going with this version of the API and not having two methods? If so feel free to merge.

**Pull request merged**